packages/web-app/src/reducers/CaveReducer.js

```diff
@@ -4,27 +4,24 @@ import {
   FETCH_CAVE_LOADING
 } from '../actions/Cave/GetCave';
 
+import { POST_DESCRIPTION_SUCCESS } from '../actions/Description/CreateDescription';
+import { UPDATE_DESCRIPTION_SUCCESS } from '../actions/Description/UpdateDescription';
+
 const initialState = {
-  data: {
-    cave: {
-      id: 1,
-      depth: 0,
-      development: 0,
-      interest: 0,
-      progression: 0,
-      access: 0,
-      name: 'Cave Name',
-      country: 'Country',
-      region: 'Region',
-      city: 'City',
-      author: 'Author'
-    },
-    entries: []
-  },
+  data: {},
   loading: false,
   error: null
 };
 
+function arrFindReplaceOrAdd(arr, test, replacement) {
+  if (!Array.isArray(arr)) return arr;
+  const newArr = arr.slice(); // Create a copy for immutability
+  const index = newArr.findIndex(test);
+  if (index < 0) newArr.push(replacement);
+  else newArr.splice(index, 1, replacement);
+  return newArr;
+}
+
```

Comment:
This can be very useful in others reducers too: I suggest putting it in a `/reducers/utils.js` file

Reply:
Modified